docs/source/vllm_integration.md

```diff
@@ -1,14 +1,72 @@
-# vLLM Integration
+# vLLM Integration 
 
-<Tip warning={true}>
+# 🎬 Flashback: why do we need to use vLLM in online methods?
 
-Section under construction. Feel free to contribute!
+Online methods such as GRPO or Online DPO require the model to generate completions. Because these completions are used to compute the reward signal, they need to be generated at regular intervals during training. This is typically done every `gradient_accumulation_steps * num_iterations` steps, where `num_iterations` is the number of iterations between two gradient updates. Now, the problem is that generating completions is a time-consuming process, especially when using large models. The reason of the time-consuming nature of the generation process is by default, this is done using the [(unwrapped)model's `generate` method](https://github.com/huggingface/trl/blob/f3e8c2304428ef16e9ae5de9e5741ed84d533b7b/trl/trainer/grpo_trainer.py#L965C39-L965C66), (ofcourse if `use_vllm` is set to False). This `unwrapped_model.generate` method is thechnically a synchronous function, meaning that it will block the execution of the program until the generation is complete. This can lead to inefficiencies, especially when generating large batches of completions or when using large models. Therefore, the generation process can become a bottleneck in the training process, leading to longer training times and reduced efficiency. So this is why we need to think of a better way to do this which is using vLLM for faster generation. 
```

Comment:
What we name `num_iterations` in GRPO isn't the number of iterations between two gradient updates. You should use another term to avoid confusion

Reply:
> meaning that it will block the execution of the program until the generation is complet

I'm not sure why you mention this? Put like this, it seems to imply that this is the major limitation that will eliminate with vLLM.